packages/forms/src/UIForm/fields/ListView/ListView.component.test.js

```diff
@@ -0,0 +1,285 @@
+import { mount, shallow } from 'enzyme';
+import keycode from 'keycode';
+import ListView from './ListView.component';
+
+jest.useFakeTimers();
+
+function filter(wrapper, searchCriteria) {
+	wrapper.find('.tc-listview-header button').at(0).simulate('click');
+	const event = { target: { value: searchCriteria } };
+	wrapper.find('.tc-listview-header input').at(0).simulate('change', event);
+	jest.runAllTimers();
+}
+
+describe('ListView field', () => {
+	let props;
+	const schema = {
+		description: 'This is the ListView field',
+		disabled: false,
+		emptyLabel: 'No element',
+		noResultLabel: 'No element matching your filter',
+		placeholder: 'Search filter',
+		required: true,
+		title: 'Countries',
+		titleMap: [
+			{ name: 'Afghanistan', value: 'country1' },
+			{ name: 'Albania', value: 'country2' },
+			{ name: 'Algeria', value: 'country3' },
+			{ name: 'Andorra', value: 'country4' },
+		],
+	};
+	const alternativeSchema = {
+		description: 'This is the ListView field',
+		disabled: true,
+		emptyLabel: 'Zero items !',
+		noResultLabel: 'Zero item for your filter !',
+		placeholder: 'Search this',
+		required: true,
+		title: 'Some countries',
+		titleMap: [
+			{ name: 'Afghanistan', value: 'country1' },
+			{ name: 'Albania', value: 'country2' },
+			{ name: 'Algeria', value: 'country3' },
+			{ name: 'Andorra', value: 'country4' },
+			{ name: 'Angola', value: 'country5' },
+			{ name: 'Anguilla', value: 'country6' },
+		],
+	};
+	const noItemsSchema = {
+		description: 'This is the ListView field',
+		disabled: false,
+		emptyLabel: 'No element',
+		noResultLabel: 'No element matching your filter',
+		placeholder: 'Search filter',
+		required: true,
+		title: 'Countries',
+		titleMap: [],
+	};
+
+	beforeEach(() => {
+		props = {
+			id: 'my-list-view',
+			isValid: true,
+			errorMessage: 'This is wrong',
+			onChange: jest.fn(),
+			onFinish: jest.fn(),
+			schema,
+			value: ['country2', 'country3'],
+		};
+	});
+
+	describe('render', () => {
+		it('should render ListView', () => {
+			// when
+			const wrapper = shallow(<ListView {...props} />);
+
+			// then
+			expect(wrapper.node).toMatchSnapshot();
+		});
+
+		it('should render no items message', () => {
+			// when
+			const wrapper = mount(<ListView {...props} schema={noItemsSchema} />);
+
+			// then
+			expect(wrapper.find('.tc-listview').at(0).node).toMatchSnapshot();
+		});
+	});
+
+	describe('componentWillReceiveProps', () => {
+		it('should update items on props.value change', () => {
+			// given
+			const node = document.createElement('div');
+			// eslint-disable-next-line react/no-render-return-value
+			const instance = ReactDOM.render(<ListView {...props} />, node);
+			expect(instance.state).toMatchSnapshot();
+
+			const allValues = props.schema.titleMap.map(option => option.value);
+
+			// when : trigger a props update
+			ReactDOM.render(<ListView {...props} value={allValues} />, node);
+
+			// then
+			expect(instance.state).toMatchSnapshot();
```

Comment:
This too hard to check if this tests is working.
Please find a better way to make this assertion.
I don't see why instance should have changed here, I must read both snapshots.
May be you should check previous state and next state ?